src/fu-engine.c

```diff
@@ -6289,6 +6289,48 @@ fu_engine_ensure_client_certificate (FuEngine *self)
 	g_debug ("client certificate exists and working");
 }
 
+static void
+fu_engine_add_firmware_gtypes (FuEngine *self)
+{
+	fu_engine_add_firmware_gtype (self, "raw", FU_TYPE_FIRMWARE);
+	fu_engine_add_firmware_gtype (self, "dfu", FU_TYPE_DFU_FIRMWARE);
+	fu_engine_add_firmware_gtype (self, "fmap", FU_TYPE_FMAP_FIRMWARE);
+	fu_engine_add_firmware_gtype (self, "ihex", FU_TYPE_IHEX_FIRMWARE);
+	fu_engine_add_firmware_gtype (self, "srec", FU_TYPE_SREC_FIRMWARE);
+}
```

Comment:
a static function only ever called only once for 5 different void functions?  Seems like it should just be in the caller function.

Reply:
oh never mind, It's two code paths now, I missed that. sorry for the noise.